### DISCUSSION TITLE
Creation of a .shp file containing the produced path

### DISCUSSION THREAD

**Discussion by helix22** [Fields2Cover/Fields2Cover#110] [Q&A]
Hello again! After exporting the path to csv, transforming the values to lat/long coordinates, removing duplicate entries, importing the points in qgis and converting them to path, I get the following image (Capture). This is fairly reasonable as any curvature data is lost within the process of acquiring only the certain points from the csv file. However, did anyone come up with an idea on how to maintain the curvature? The ultimate goal is to have a final .shp file, containing the path that we see in the exported from f2c image (Capture_f2c). If you have any other process in mind, please feel free to share! :)

![Capture_f2c](https://github.com/Fields2Cover/Fields2Cover/assets/45968661/55cfbd30-e49d-4c95-8508-10849043f16e)
--
![Capture](https://github.com/Fields2Cover/Fields2Cover/assets/45968661/3e53621f-7ccf-4866-ba9b-6204510be407)

**Comment by Gonzalo-Mier** [comment]
Hi Mike,

How do you transform those points to lat/lon? Are you doing the
transformation with f2c or with other method? Probably you are rounding the
lat/lon values by not getting enough decimals.

**Reply by helix22** [reply]
Hello! I actually use the short code attached below, prior to which I 've replaced spaces with commas in each row of the output from f2c path.csv file. Hmm you are right that could be a reason.. However, if I take the path.csv file prior to converting it to latlon coords and I remove duplicate entries in the first 2 columns, I still get the same number of unique values as after the conversion. So the output points must be of the same number.. If I am not missing something here of course!

```python
import csv
import pyproj

def utm_to_latlon(easting, northing, zone_number=34, northern_hemisphere=True):
    """
    Convert UTM coordinates to latitude and longitude.

    Args:
        easting (float): Easting coordinate in meters.
        northing (float): Northing coordinate in meters.
        zone_number (int): UTM zone number (default is 34).
        northern_hemisphere (bool): True if in northern hemisphere, False otherwise (default is True).

    Returns:
        tuple: Latitude and longitude in degrees.
    """
    # Define UTM projection
    utm_proj = pyproj.Proj(proj='utm', zone=zone_number, ellps='WGS84', north=northern_hemisphere)
  
    # Convert UTM coordinates to latitude and longitude
    lon, lat = utm_proj(easting, northing, inverse=True)
   
    return lat, lon

# Read coordinates from path.csv and convert them to latitude and longitude
with open('path_UTM.csv', 'r') as infile, open('path_LATLON.csv', 'w', newline='') as outfile:
    reader = csv.reader(infile)
    writer = csv.writer(outfile)
    
    for row in reader:
        easting, northing = float(row[0]), float(row[1])
        latitude, longitude = utm_to_latlon(easting, northing)
        writer.writerow([latitude, longitude])
``` 

For the record, **path_UTM.csv** first 5 rows look like:
`<html>
<body>
<!DOCTYPE html>

<html>
<head>
	
	
	<title></title>
	<meta name="generator" content="LibreOffice 7.3.7.2 (Linux)"/>
	<style type="text/css">
		body,div,table,thead,tbody,tfoot,tr,th,td,p { font-family:"Liberation Sans"; font-size:x-small }
		a.comment-indicator:hover + comment { background:#ffd; position:absolute; display:block; border:1px solid black; padding:0.5em;  } 
		a.comment-indicator { background:red; display:inline-block; border:1px solid black; width:0.5em; height:0.5em;  } 
		comment { display:none;  } 
	</style>
	
</head>

<body>

572137 | 4460460 | 0 | 1.5708 | 1 | 16.2879 | 1 | 1
-- | -- | -- | -- | -- | -- | -- | --
572137 | 4460480 | 0 | 1.5708 | 1 | 0 | 1 | 1
572137 | 4460480 | 0 | 1.5708 | 1 | 0.00999999 | 1 | 2
572137 | 4460480 | 0 | 1.5758 | 1 | 0.00999999 | 1 | 2
572137 | 4460480 | 0 | 1.5808 | 1 | 0.00999999 | 1 | 2


</body>

</html>

</body>
</html>
`

and **path_LATLON.csv** first 5 rows, after removing the duplicate values, look like: 
<html>
<body>
<!DOCTYPE html>

<html>
<head>
	
	
	<title></title>
	<meta name="generator" content="LibreOffice 7.3.7.2 (Linux)"/>
	<style type="text/css">
		body,div,table,thead,tbody,tfoot,tr,th,td,p { font-family:"Liberation Sans"; font-size:x-small }
		a.comment-indicator:hover + comment { background:#ffd; position:absolute; display:block; border:1px solid black; padding:0.5em;  } 
		a.comment-indicator { background:red; display:inline-block; border:1px solid black; width:0.5em; height:0.5em;  } 
		comment { display:none;  } 
	</style>
	
</head>

<body>

40.29152522 | 21.84870945
-- | --
40.29170538 | 21.84871171
40.29170547 | 21.84869994
40.29179555 | 21.84870107
40.29179564 | 21.84868931


</body>

</html>

</body>
</html>

**Comment by helix22** [comment]
how do I bump here? thanks! (:
